mutation_writer/multishard_writer.cc

```diff
@@ -131,16 +132,26 @@ future<stop_iteration> multishard_writer::handle_mutation_fragment(mutation_frag
     auto f = make_ready_future<>();
     if (mf.is_partition_start()) {
         _consumed_partitions++;
-        if (unsigned shard = shard_for_mf(mf); shard != _current_shard) {
-            _current_shard = shard;
-            if (!bool(_shard_writers[shard])) {
-                f = make_shard_writer(shard);
+        if (auto shards = shard_for_mf(mf); shards != _current_shards) {
+            _current_shards = shards;
+            // FIXME: Keep effective_replication_map ptr as long as we use those shards.
+            for (auto shard : shards) {
+                if (!bool(_shard_writers[shard])) {
+                    f = f.then([this, shard] {
+                        return make_shard_writer(shard);
+                    });
+                }
             }
         }
     }
     return f.then([this, mf = std::move(mf)] () mutable {
-        assert(_current_shard != -1u);
-        return _queue_reader_handles[_current_shard]->push(std::move(mf));
+        assert(!_current_shards.empty());
+        if (_current_shards.size() == 1) [[likely]] {
+            return _queue_reader_handles[_current_shards[0]]->push(std::move(mf));
+        }
+        return seastar::parallel_for_each(_current_shards, [this, mf = std::move(mf)] (unsigned shard) {
+            return _queue_reader_handles[shard]->push(mutation_fragment_v2(*_s, mf.permit(), mf));
+        });
     }).then([] {
```

Comment:
multishard_writer should be avoided when using tablets, but I suppose we'll have to defer its removal too.